### PR TITLE
gtest: add tap output support in gtest

### PR DESCRIPTION
**Pull request by miked-mellanox**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1236/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/649/ for details (Mellanox internal link).

**Comment by miked-mellanox** [comment]
@yosefe - could you please analyze gtest failures? 
now it is running tests in a random fashion.
failures unrelated to the PR.

File: src/ucs/gtest/main.cc [hunk 1]
```diff
@@ -44,6 +44,19 @@ static void modify_config_for_valgrind(const char *name, const char *value)
 
 int main(int argc, char **argv) {
     ::testing::InitGoogleTest(&argc, argv);
+
+    char *str = getenv("GTEST_TAP");
+    /* Append TAP Listener */
+    if (str) {
+        if (0 < strtol(getenv("GTEST_TAP"), NULL, 0)) {
```
Comment (yosefe) [inline src/ucs/gtest/main.cc]: can we reuse 'str' instead of calling getenv all the time?
e.g 
```
int gtest_tap = strtol(str, NULL, 0)
```

**Comment by shamisp** [comment]
is this something that you want to use for gtest / jenkins integration ? What is the main goal.

**Comment by shamisp** [comment]
autobot build

**Comment by miked-mellanox** [comment]
@shamisp - yes, jenkins will show tests statistics based on tap format.
also, now tests are executed in random order.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1240/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/653/ for details (Mellanox internal link).

**Comment by miked-mellanox** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1243/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/656/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1244/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/657/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1249/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/662/ for details (Mellanox internal link).

**Comment by miked-mellanox** [comment]
reviewed by @yosefe , verbal ok

**Comment by yosefe** [comment]
:+1:

**Pull request merged**